astropy/io/misc/ecsv.py

```diff
@@ -209,57 +216,85 @@ class ECSVEngine:
         Dictionary mapping engine names to their respective engine classes.
     """
 
```

Comment:
We were talking about `cached_property` compatibility with `__slots__`.

```suggestion

    __slots__: tuple[str, ...] = ("name", "format", "engines", "__dict__")

```

The advantage of slots is speed and memory use. But the disadvantage now is that `__dict__` means it can store any attribute through attribute assignment. This is where `frozen=True` dataclasses comes in... it protects assignment to the dict.